### PR TITLE
Update node-gyp so yarn install also works with Visual Studio 2022

### PR DESCRIPTION
**Pull request by CumpsD**
## Summary

When trying to do `yarn install` on Windows, it fails trying to build the c++ modules. If you install the latest VS (2022) the included node-gyp (7.x) does not support it yet. Update to 8.x so it works.

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
     "eslint-plugin-simple-import-sort": "7.0.0",
     "jest": "26.6.3",
     "lerna": "4.0.0",
+    "node-gyp": "8.4.1",
```
Comment (NullSoldier) [inline package.json:54]: We shouldn't add this here, it should be added to the project that requires node-gyp. Which project are you seeing that is trying to use node-gyp to build?

**Comment by NullSoldier** [comment]
Can you post the error you're getting without this?

**Comment by CumpsD** [comment]
@NullSoldier 

```
[4/4] Building fresh packages...
[1/7] ⠈ mrmr
[-/7] ⠈ waiting...
[-/7] ⠈ waiting...
[-/7] ⠈ waiting...
error c:\Work\CumpsD\ironfish\node_modules\mrmr: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments:
Directory: c:\Work\CumpsD\ironfish\node_modules\mrmr
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.0.0 | win32 | x64
gyp info find Python using Python version 2.7.18 found at "C:\Python27\python.exe"
gyp ERR! find VS
gyp ERR! find VS msvs_version was set from command line or npm config
gyp ERR! find VS - looking for Visual Studio version 2022
gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
gyp ERR! find VS unknown version "undefined" found at "C:\Program Files\Microsoft Visual Studio\2022\Community"
gyp ERR! find VS checking VS2019 (16.11.32002.261) found at:
gyp ERR! find VS "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools"
gyp ERR! find VS - found "Visual Studio C++ core features"
gyp ERR! find VS - found VC++ toolset: v142
gyp ERR! find VS - found Windows SDK: 10.0.19041.0
gyp ERR! find VS - msvs_version does not match this version
gyp ERR! find VS could not find a version of Visual Studio 2017 or newer to use
gyp ERR! find VS looking for Visual Studio 2015
gyp ERR! find VS - not found
gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
gyp ERR! find VS
gyp ERR! find VS valid versions for msvs_version:
gyp ERR! find VS - "2019"
gyp ERR! find VS - "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools"
gyp ERR! find VS
gyp ERR! find VS **************************************************************
gyp ERR! find VS You need to install the latest version of Visual Studio
gyp ERR! find VS including the "Desktop development with C++" workload.
gyp ERR! find VS For more information consult the documentation at:
gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
gyp ERR! find VS **************************************************************
gyp ERR! find VS
gyp ERR! configure error
gyp ERR! stack Error: Could not find any Visual Studio installation to use
gyp ERR! stack     at VisualStudioFinder.fail (c:\Work\CumpsD\ironfish\node_modules\@npmcli\run-script\node_modules\node-gyp\lib\find-visualstudio.js:121:47)
gyp ERR! stack     at c:\Work\CumpsD\ironfish\node_modules\@npmcli\run-script\node_modules\node-gyp\lib\find-visualstudio.js:74:16
gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (c:\Work\CumpsD\ironfish\node_modules\@npmcli\run-script\node_modules\node-gyp\lib\find-visualstudio.js:351:14)
gyp ERR! stack     at c:\Work\CumpsD\ironfish\node_modules\@npmcli\run-script\node_modules\node-gyp\lib\find-visualstudio.js:70:14
gyp ERR! stack     at c:\Work\CumpsD\ironfish\node_modules\@npmcli\run-script\node_modules\node-gyp\lib\find-visualstudio.js:372:16
gyp ERR! stack     at c:\Work\CumpsD\ironfish\node_modules\@npmcli\run-script\node_modules\node-gyp\lib\util.js:54:7
gyp ERR! stack     at c:\Work\CumpsD\ironfish\node_modules\@npmcli\run-script\node_modules\node-gyp\lib\util.js:33:16
gyp ERR! stack     at ChildProcess.exithandler (node:child_process:333:5)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at maybeClose (node:internal/child_process:1067:16)
gyp ERR! System Windows_NT 10.0.22000
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "c:\\Work\\CumpsD\\ironfish\\node_modules\\@npmcli\\run-script\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd c:\Work\CumpsD\ironfish\node_modules\mrmr
```

It's using node-gyp@7.1.2, which doesn't support 2022

**Comment by NullSoldier** [comment]
It works on mac.

**Review by dguenther — APPROVED** [review]
I built this in a Windows Sandbox using VS 2022 and confirmed that it runs as expected.

**Pull request merged**